geps/gep-1651.md

```diff
@@ -0,0 +1,218 @@
+
+const (
+  GatewayReachabilityPublic  GatewayReachability = "public"
+  GatewayReachabilityCluster GatewayReachability = "cluster"
+)
+
+type GatewaySpec struct {
+  // Infrastructure defines infrastructure level attributes about this Gateway instance.
+  Infrastructure GatewayInfrastructure `json:"infrastructure"`
+  // ...
+}
+type GatewayInfrastructure struct {
+  // AddressScope allows the Gateway to specify the accessibility of it's addresseses
+  Reachability GatewayReachability `json:"reachability"`
+}
+```
+
+### YAML
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: public
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+### Semantics
+
+#### Unsupported reachability & address values
+
+If a Gateway is unable provide an address for the desired reachability it MUST set the condition `Accepted` to `False`
+
+### Status.Addresses
+
+When 'reachability' is specified in the Gateway's `spec` implementations MUST populate the `status.addresses` with
+an address with the desired reachability semantic.
+
+## Examples
+
+#### 1. Request a GatewayAddress that is only reachable within the same cluster
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+#### 2. Request a GatewayAddress with a specific reachability and address
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+  addresses:
+  - value: 10.0.0.8
+```
+#### 3. Request a GatewayAddress that is publically reachable
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: public
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+#### 4. Request a GatewayAddress that is a cloud provider's VPC
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: some.cloud.dev/vpc
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+## Alternatives
+
+### Introducing new GatewayAddress Types
+
+We could introduce additional `AddressTypes` (ie. `ClusterLocalIPAddress`) but
+this would lead to a combinatorial explosion as new dimensions (ie. IPv6) are
+introduced.
+
+See examples in this comment: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1451246877
+
+### Scope/reachability field on GatewayAddress
+
+This would allow Gateways to have multiple scopes.
+
+See pros/cons here: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1486271913
```

Comment:
Mind including these in the GEP itself so it's easier to read? Same suggestion applies to link above.

Reply:
done